lib/chef/provider/package/macports.rb

```diff
@@ -3,7 +3,7 @@ class Provider
     class Package
       class Macports < Chef::Provider::Package
 
-        provides :macports_package, os: "mac_os_x"
+        provides :macports_package, os: "darwin"
```

Comment:
Hmmmm....  okay, I seem to have not restricted yum_package, apt_package, etc, so I guess I agreed with you back when I merged in the provider resolver stuff (although I coulda sworn I decided to do the other thing).